Add PENDING enrollment status for SIS enrollments

PasiSa opened the pull request:
The PENDING enrollment status is used for courses with enrollment
questionnaires, when students are enrolled from SIS. PENDING students
cannot start the course tasks before completing enrollment
questionnaire.

Fixes #981.

# Description

**What?**

As described above, to support use case with SIS enrollments with enrollment questionnaire.

As discussed in our teams channel, the current way of presenting the enrolled / removed student counts is not ideal. For now, the PR proposes replacing texts for enrolled/removed by active/inactive as a quick fix. The proposed wording may not be very clear either, but the former wording becomes incorrect when PENDING status is introduced. I can start working on a better way of representing the enrollment status counts, but that would be in separate PR and commit.

**Why?**

[ANSWER HERE]

**How?**

[ANSWER HERE]

Fixes #<issue number>


# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)


**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [ ] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [ ] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [ ] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

Mankro reviewed (changes requested):
Good! There are a couple of small mistakes to fix. You didn't update all uses of the `find_enrollment_exercise()` method.

Note to production: this PR adds new migrations.

Concerning the "total enrolled students" line in the participants page, I vote for improving the presentation (in the next pull request).

PasiSa commented:
Updated the PR. In addition to addressing the comments, I now added PENDING status also to non-SIS courses, as discussed with Juha, and fixed one other error (upgrading enrollement from PENDING to ACTIVE) that I had somehow missed earlier. Those are in separate commit that should be squashed before merge.

Mankro reviewed (approved):
Great! I will squash the commits myself and merge this now.
